lib/cfn/git_client.rb

```diff
@@ -39,13 +40,17 @@ def show_defaults(name, ref = { branch: 'master' })
       { name: name, data: JSON.parse(body) }
     end
 
-    def list_templates(filter = '*', ref = { branch: 'master' })
+    def list_templates(filter = '**/*', ref = { branch: 'master' })
       reset_hard_ref(ref)
       path_with_glob = workdir_path("templates/#{filter}")
-      files = Dir.glob(path_with_glob)
-      files.map do |path|
-        ext = File.extname(path)
-        { name: File.basename(path, ext) }
+      base_path = Pathname.new(workdir_path("templates/"))
+      files = Dir.glob(path_with_glob).map { |p| Pathname.new(p) }
+      template_files = files.select { |path| path.extname.match(/\A(yml)|(yaml)|(json)\z/) }
```

Comment:
I thought about that, but I did the regex so that I didn't have to worry about wonky case sensitivity.